pkg/compose/watch_test.go

```diff
@@ -0,0 +1,29 @@
+package compose
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
+)
+
+func Test_debounce(t *testing.T) {
```

Comment:
You might take a look at Go libs out there to help with controlling timers deterministically in tests, as this is likely to become a problematic/failing test in CI otherwise

Tilt uses this library fairly extensively: https://github.com/jonboulle/clockwork

It provides a clock, which is either the real one (i.e. pass calls through to `time.Time`) or a fake one where you can control when/how it advances